pkg/storage/db.go

```diff
@@ -151,10 +152,35 @@ func (db *DB) Init(ctx context.Context, postgresConfig *PostgresConfig) error {
 
 		}
 		initerr = db.RunScript(ctx, createScript)
+
+		if initerr == nil {
+			db.startPartitionGeneratorTimer(ctx)
+		}
 	})
 	return initerr
 }
 
+func (db *DB) startPartitionGeneratorTimer(ctx context.Context) {
+	// run every 12 hours, with the assumption that partition tables will never be
+	// this granular, thus partition tables are created well in advance of their need
+	ticker := time.NewTicker(12 * time.Hour)
```

Comment:
Question: There are a few edge cases we'll need to consider for a Lambda environment. Using the question prefix since I'm looking for your thoughts.

Official runtime docs: https://docs.aws.amazon.com/lambda/latest/dg/running-lambda-code.html

In the most degenerate case, Lambda will create a new VM to handle a request and then immediately pause that VM. Any VM re-use is not guaranteed and is only an optimization we may get. In the degenerate case, this timer will never fire because the VM doesn't exist more more than the milliseconds it takes to resolve one function call. In the case where a VM is reused, I _think_ the event will fire after looking at the Go code that manages time based events. 

Another thing to consider is that the instance count of this function is not under our control and _will_ be scaled out to meet demands on the function invocation.  The default limit is [1000 concurrent executions](https://docs.aws.amazon.com/lambda/latest/dg/limits.html) which I'm taking to mean 1000 instances of the function running in isolated VMs. We do have some control over this but by having this as a part of the DB connection init means that it will run as part of all instances of all functions we run. There are only three after this PR but we expect more to be added over time for different query types. For us, this might mean several hundred instances racing to lock the partition tables and run DDL statements. This might not be a problem but concurrent DDL in other systems has caused me problems in the past so I get nervous when I see it.

All of this is to say that Lambda isn't really intended to support this kind of behavior. What's the driving motivator to make this time based instead of trigger based?

If it helps, Lambda can be triggered on a [schedule via CloudWatch](https://docs.aws.amazon.com/lambda/latest/dg/tutorial-scheduled-events-schedule-expressions.html) and we have an internal cron-like system that can provide this functionality as well.

Reply:
Super good points.  I was not thinking about the execution conditions this code would find itself in under lambda.  Now that I _am_ thinking about it... :)

@zlozano and I talked over the endpoint trigger as the mechanism for ensuring pre-emptive existence of the required child partition table, and had some worries about it.  Specifically, how to ensure the cronman schedule was correct.

That said, given it's lambda, maybe we _should_ run cronman schedule daily to hit a trigger endpoint, and just let the service decide if it's really time to create the next partition or not.

I'm inclined to go back to that ^^^ trigger solution now.

Reply:
@kconwayatlassian and @zlozano and I talked, each independently, and both have great points.  We shall talk in person and record the result here.  Meantime, I'm going to improve test coverage.

Reply:
Result if conversation is recorded in the PR description.